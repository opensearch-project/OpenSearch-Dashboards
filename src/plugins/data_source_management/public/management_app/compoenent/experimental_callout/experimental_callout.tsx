/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';
import { DocLinksStart } from 'opensearch-dashboards/public';
import { i18n } from '@osd/i18n';
import { FormattedMessage } from '@osd/i18n/react';

export const ExperimentalCallout = ({ docLinks }: { docLinks: DocLinksStart }) => {
  return (
    <>
      <EuiCallOut
        title={i18n.translate('dataSourcesManagement.experimentalFeatureCallout.title', {
          defaultMessage: 'Experimental Feature',
        })}
        iconType="iInCircle"
        data-test-subj="data-source-experimental-call"
      >
        <p>
          {
            <FormattedMessage
              id="dataSourcesManagement.experimentalFeatureCallout.description"
              defaultMessage="This feature is experimental and should not be used in a production environment. Any index patterns, visualization, and observability panels will be impacted if the feature is deactivated. For more information see "
            />
          }
          <EuiLink href={docLinks.links.noDocumentation.dataSource.guide} target="_blank">
            {
              <FormattedMessage
                id="dataSourcesManagement.experimentalFeatureCallout.documentationText"
                defaultMessage="Data Source Documentation"
              />
            }
          </EuiLink>
          {
            <FormattedMessage
              id="dataSourcesManagement.experimentalFeatureCallout.feedbackText"
              defaultMessage="To leave feedback, visit "
            />
          }
          <EuiLink href={docLinks.links.noDocumentation.openSearchForum} target="_blank">
            {
              <FormattedMessage
                id="dataSourcesManagement.experimentalFeatureCallout.openForumText"
                defaultMessage="forum.opensearch.org"
              />
            }
          </EuiLink>
        </p>
      </EuiCallOut>
      <EuiSpacer size="m" />
    </>
  );
};
