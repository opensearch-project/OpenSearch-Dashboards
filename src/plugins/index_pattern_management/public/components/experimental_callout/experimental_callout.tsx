/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { i18n } from '@osd/i18n';
import { EuiCallOut, EuiSpacer, EuiText } from '@elastic/eui';

export const ExperimentalCallout = () => {
  return (
    <>
      <EuiCallOut
        title={TITLE}
        iconType="alert"
        color="warning"
        data-test-subj="index-pattern-experimental-callout"
      >
        <EuiText data-test-subj="index-pattern-experimental-callout-text">
          <p>
            {DESCRIPTION_FIRST_PART}
            <b>{DATASOURCE_CONNECTION}</b>
            {DESCRIPTION_SECOND_PART}
            <b>{DEFAULT}</b>
            {DESCRIPTION_THIRD_PART}
          </p>
        </EuiText>
      </EuiCallOut>
      <EuiSpacer size="m" />
    </>
  );
};

const TITLE = i18n.translate('indexPatternManagement.experimentalFeatureCallout.title', {
  defaultMessage: 'Experimental feature active',
});

const DESCRIPTION_FIRST_PART = i18n.translate(
  'indexPatternManagement.experimentalFeatureCallout.descriptionPartOne',
  {
    defaultMessage: 'The experimental feature ',
  }
);
const DATASOURCE_CONNECTION = i18n.translate(
  'indexPatternManagement.experimentalFeatureCallout.datasourceConnection',
  {
    defaultMessage: 'Data Source Connection ',
  }
);
const DESCRIPTION_SECOND_PART = i18n.translate(
  'indexPatternManagement.experimentalFeatureCallout.descriptionPartTwo',
  {
    defaultMessage:
      'is active. To create an index pattern without using data from an external source, use ',
  }
);
const DEFAULT = i18n.translate('indexPatternManagement.experimentalFeatureCallout.default', {
  defaultMessage: 'default',
});
const DESCRIPTION_THIRD_PART = i18n.translate(
  'indexPatternManagement.experimentalFeatureCallout.descriptionPartThree',
  {
    defaultMessage:
      '. Any index pattern created using an external data source will result in an error if the experimental feature is deactivated.',
  }
);
