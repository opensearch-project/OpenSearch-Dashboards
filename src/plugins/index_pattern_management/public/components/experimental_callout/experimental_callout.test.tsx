/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { mount } from 'enzyme';
import { ExperimentalCallout } from './experimental_callout';

const titleIdentifier = '.euiCallOutHeader__title';
const descriptionIdentifier = '[data-test-subj="index-pattern-experimental-callout-text"]';
const expectedTitleText = 'Experimental feature active';
const expectedDescriptionText =
  'The experimental feature Data Source Connection is active. To create an index pattern without using data from an external source, use default. Any index pattern created using an external data source will result in an error if the experimental feature is deactivated.';

describe('Index pattern experimental callout component', () => {
  test('should render normally', () => {
    const component = mount(<ExperimentalCallout />);
    const titleText = component.find(titleIdentifier).text();
    const descriptionText = component.find(descriptionIdentifier).last().text();

    expect(titleText).toBe(expectedTitleText);
    expect(descriptionText).toBe(expectedDescriptionText);
    expect(component).toMatchSnapshot();
  });
});
