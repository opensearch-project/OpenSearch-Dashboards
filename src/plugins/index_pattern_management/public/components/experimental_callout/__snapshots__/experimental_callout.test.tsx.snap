// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index pattern experimental callout component should render normally 1`] = `
<ExperimentalCallout>
  <EuiCallOut
    color="warning"
    data-test-subj="index-pattern-experimental-callout"
    iconType="alert"
    title="Experimental feature active"
  >
    <div
      className="euiCallOut euiCallOut--warning"
      data-test-subj="index-pattern-experimental-callout"
    >
      <div
        className="euiCallOutHeader"
      >
        <EuiIcon
          aria-hidden="true"
          className="euiCallOutHeader__icon"
          color="inherit"
          size="m"
          type="alert"
        >
          <span
            aria-hidden="true"
            className="euiCallOutHeader__icon"
            color="inherit"
            data-euiicon-type="alert"
            size="m"
          />
        </EuiIcon>
        <span
          className="euiCallOutHeader__title"
        >
          Experimental feature active
        </span>
      </div>
      <EuiText
        color="default"
        size="s"
      >
        <div
          className="euiText euiText--small"
        >
          <EuiTextColor
            color="default"
            component="div"
          >
            <div
              className="euiTextColor euiTextColor--default"
            >
              <EuiText
                data-test-subj="index-pattern-experimental-callout-text"
              >
                <div
                  className="euiText euiText--medium"
                  data-test-subj="index-pattern-experimental-callout-text"
                >
                  <p>
                    The experimental feature 
                    <b>
                      Data Source Connection 
                    </b>
                    is active. To create an index pattern without using data from an external source, use 
                    <b>
                      default
                    </b>
                    . Any index pattern created using an external data source will result in an error if the experimental feature is deactivated.
                  </p>
                </div>
              </EuiText>
            </div>
          </EuiTextColor>
        </div>
      </EuiText>
    </div>
  </EuiCallOut>
  <EuiSpacer
    size="m"
  >
    <div
      className="euiSpacer euiSpacer--m"
    />
  </EuiSpacer>
</ExperimentalCallout>
`;
